### PR TITLE
belindas-closet-nestjs_5_101_add-testing-configurations-and-auth-test

### PR DESCRIPTION
**Pull request by IsaacJrTypes**
This pull request adds testing configurations and an authentication test. Fixed all issues with unsigned commits. 
CleanShot 2024-04-29 at 19 48 04@2x

### CODE REVIEW

**Review by taylorpapke — APPROVED** [review]
Works on my end:
Screenshot 2024-04-30 at 2 33 52 AM

**Review by brinkbrink — APPROVED** [review]
Screenshot 2024-05-02 at 12 44 23 PM

Working as expected. And I'm assuming result is not used because it will be used in what is currently an unfinished test:

Screenshot 2024-05-02 at 12 46 25 PM

Happy to see you got your commit signing figured out. So I'll only repeat half of what I said on the last PR: don't forget to include a description on how to test the changes you've made and be explicit about what the reviewer should expect to see and why. 

Thanks again, this is really important work.

**Review by keiffer213 — APPROVED** [review]
Look good! I'm getting the same results as the other approvals as well! 

image
image
image

**Review by intisarosman1 — APPROVED** [review]
All tests passed. Looks good!
Screenshot 2024-05-10 082342

File: src/auth/controllers/auth.controller.spec.ts
Comment (gitbiruk2010) [inline src/auth/controllers/auth.controller.spec.ts:1]: It’s a cool feature to use mocking with Jest to test the AuthController. Mocking allows us to isolate the unit under test (in this case, the AuthController) and test its interactions without relying on the actual implementation of dependencies like AuthService.

**Review by ogarmela — APPROVED** [review]

**Review by Seiyaroo — APPROVED** [review]
Tests passing, looks good.

**Comment by taylorpapke** [comment]
@IsaacJrTypes Should we go ahead and close this PR or is there something else that is needed from this and/or the branch?

**Comment by IsaacJrTypes** [comment]
@taylorpapke I had to update the file since there were merge conflicts. We are good to close this pull request

**Pull request merged**